nrf_wifi/fw_if/umac_if/src/event.c

```diff
@@ -885,19 +890,69 @@ static enum nrf_wifi_status umac_process_sys_events(struct nrf_wifi_fmac_dev_ctx
 		break;
 	case NRF_WIFI_EVENT_RADIOCMD_STATUS:
 		umac_status = ((struct nrf_wifi_umac_event_err_status *)sys_head);
-		def_dev_ctx->radio_cmd_status = umac_status->status;
-		def_dev_ctx->radio_cmd_done = true;
+		def_dev_ctx_rt->radio_cmd_status = umac_status->status;
+		def_dev_ctx_rt->radio_cmd_done = true;
 		status = NRF_WIFI_STATUS_SUCCESS;
 		break;
 #endif /* CONFIG_NRF700X_RADIO_TEST */
+#ifdef CONFIG_NRF700X_RAW_DATA_TX
+	case NRF_WIFI_EVENT_RAW_TX_DONE:
+		status = nrf_wifi_fmac_rawtx_done_event_process(fmac_dev_ctx,
+						(struct nrf_wifi_event_raw_tx_done *)sys_head);
+		break;
+	case NRF_WIFI_EVENT_MODE_SET_DONE:
+		struct nrf_wifi_event_raw_config_mode *mode_event;
+
+		mode_event = (struct nrf_wifi_event_raw_config_mode *)sys_head;
+		if (!mode_event->status) {
+			def_dev_ctx->vif_ctx[mode_event->if_index]->mode =
+								mode_event->op_mode;
+			/**
+			 * Set the transmit queue for RAW packet transmission
+			 */
+			if (mode_event->op_mode == (NRF_WIFI_TX_INJECTION_MODE |
+						    NRF_WIFI_STA_MODE)) {
+				def_dev_ctx->tx_config.peers[MAX_PEERS].peer_id = MAX_PEERS;
+				def_dev_ctx->tx_config.peers[MAX_PEERS].if_idx =
+									mode_event->if_index;
+				def_dev_ctx->vif_ctx[mode_event->if_index]->if_type =
+									NRF_WIFI_STA_TX_INJECTOR;
```

Comment:
Better to have this as just a boolean flag in vif_ctx (i.e. vif_ctx->tx_inj = true`) instead of trying to manipulate the `if_type`. `if_type` should just be left as base modes i.e. STA, AP, or MONITOR

Reply:
No, this is not okay. - Maintaining one flag for TX injection means that we would need to maintain one flag for promiscuous as well even if we consider monitor as a full interface type. Tomorrow, we will have combinations such as station+ tx_injection + promiscuous , station + Tx_injection, Monitor + Tx_injection, Station + Promiscuous etc. Maintaining 2-3 flags and different interface flags in the code and adding it every where for checks will make the code pretty messy. What we tell the outside as default interfaces does not have to be how driver needs to maintain context. If there is a choice between maintaining one mode operation flag vs 3-4 mode operation flags. 1 flag should always be the way to go